independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/ComputingCache.java

```diff
@@ -21,18 +22,42 @@
 public class ComputingCache<K, V> {
 
     private final ConcurrentMap<K, LazyValue<V>> map;
+    private final Function<K, V> computingFunction;
 
-    private final Function<K, LazyValue<V>> function;
+    /**
+     * Note that {@link #getValue(Object)} cannot be used if no default computing function is specified.
+     */
+    public ComputingCache() {
+        this(null);
+    }
 
     public ComputingCache(Function<K, V> computingFunction) {
         this.map = new ConcurrentHashMap<>();
-        this.function = new CacheFunction(computingFunction);
+        this.computingFunction = computingFunction;
     }
 
     public V getValue(K key) {
+        return computeIfAbsent(key, computingFunction);
+    }
+
+    public V getValueIfPresent(K key) {
+        LazyValue<V> value = map.get(key);
+        return value != null ? value.getIfPresent() : null;
+    }
+
+    public V computeIfAbsent(K key,
+            Function<? super K, ? extends V> computingFunction) {
+        return computeIfAbsent(key, () -> computingFunction.apply(key));
```

Comment:
Could we avoid the lambda in runtime code?

Reply:
Of course we could ;-). Although we used lambdas even before this PR...